File: AndroidStudio/Activity2-1/app/src/main/java/mx/itesm/activity2_1/MainActivity.java
package mx.itesm.activity2_1;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class MainActivity extends AppCompatActivity implements JSONRequest.JSONRequestCallback, AdapterView.OnItemClickListener {

    ListView list;
    JSONAdapter jsonAdapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        list = (ListView) findViewById(R.id.list_view);
    }

    /*
    [{"name":"Miguel","hobby":"Netflix","age":"19","phone":"3311224466","address":"Near-by"},{"name":"Santiago","hobby":"Civilization","age":"20","phone":"3456745300","address":"Far-far-away"},{"name":"Casiel","hobby":"Dogs","age":"20","phone":"3124534623","address":"First floor"},{"name":"Sebastián","hobby":"Racism","age":"9-ish","phone":"6666666666","address":"Next to Hitler"},{"name":"Edgar","hobby":"Driving","age":"95","phone":"1001100110","address":"Around the corner"}]
     */

    public void request(View v){
        JSONRequest r = new JSONRequest(this);
        r.execute("https://api.myjson.com/bins/1b9ml7");
    }

    @Override
    public void done(JSONArray jsonArray) {
        Toast.makeText(this, "DONE JSON-ing", Toast.LENGTH_SHORT).show();
        try{
            for(int i = 0; i < jsonArray.length(); i++){
                Log.d("JSON",i+" "+jsonArray.getJSONObject(i).get("hobby"));
            }
            jsonAdapter = new JSONAdapter(jsonArray, this);
            list.setAdapter(jsonAdapter);
            list.setOnItemClickListener(this);
        }catch(JSONException json){
            json.printStackTrace();
        }
    }

    @Override
    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
        Toast.makeText(this, "Clicked!", Toast.LENGTH_SHORT).show();
        try{
            Intent i = new Intent(this, Friend.class);
            JSONObject json = (JSONObject)parent.getItemAtPosition(position);
            i.putExtra("name",json.getString("name"));
            i.putExtra("hobby",json.getString("hobby"));
            i.putExtra("age",json.getString("age"));
            i.putExtra("phone",json.getString("phone"));
            i.putExtra("address",json.getString("address"));
            Toast.makeText(this, json.getString("name"), Toast.LENGTH_SHORT).show();
            startActivity(i);
        }catch(JSONException joe){
            joe.printStackTrace();
        }
    }
}

File: Xamarin/Activity3-3/Activity3-3/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;
using System;
using System.Collections.Generic;
using Android.Runtime;
using Org.Json;

namespace Activity3_3
{
    [Activity(Label = "Activity3_3", MainLauncher = true, Icon = "@drawable/icon")]
    public class MainActivity : Activity , JSONRequest.JSONListener
    {
        Button buttonJSON;
        ListView lv;

        public void Listen(JSONArray posts)
        {
            JSONAdapter adapter = new JSONAdapter(this, posts);
            lv.Adapter = adapter;
        }

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);

            // Set our view from the "main" layout resource
            SetContentView (Resource.Layout.Main);

            buttonJSON = FindViewById<Button>(Resource.Id.button_get_json);
            lv = FindViewById<ListView>(Resource.Id.listView_json);

            buttonJSON.Click += delegate
            {
                JSONRequest request = new JSONRequest(this);
                request.Execute("http://jsonplaceholder.typicode.com/posts");
            };
        }


    }
}

